core/modules/user/user.module

```diff
@@ -998,6 +1009,54 @@ function user_account_form_validate($form, &$form_state) {
   }
 }
 
+/**
+ * Validation callback for password constraints.
+ *
+ * @see user_account_form().
+ * @see user_pass_reset_form().
+ */
+function user_password_constraints_validate($form, &$form_state) {
+  if (!array_key_exists('pass', $form_state['values'])) {
+    // User registration forms don't have a password field,
+    // if mail verification is required.
+    return;
+  }
+  $password = trim($form_state['values']['pass']);
+  if ($form['#form_id'] == 'user_profile_form' && strlen($password) == 0) {
+    // On profile form and no new password has been set.
+    return;
+  }
+  $constraints = config_get('system.core', 'password_constraints');
+  $messages = array();
+  if ($form['#form_id'] == 'user_pass_reset_form') {
+    $account = $form['#account'];
+    $username = $account->name;
+    $email = $account->mail;
+  }
+  else {
+    $username = $form_state['values']['name'];
+    $email = $form_state['values']['mail'];
+  }
+  if (strlen($password) < $constraints['minlength']) {
+    $min = $constraints['minlength'];
+    $messages[] = t('The password is too short. At least :min characters required.', array(':min' => $min));
```

Comment:
I don't believe `:` is a supported prefix for variables (I think they have to be one of: `@, %, !`)... https://api.backdropcms.org/api/backdrop/core%21includes%21bootstrap.inc/function/t

Reply:
Gosh, how did these colons sneak in...? :facepalm: